bson.cpp

```diff
@@ -1064,6 +1178,64 @@ void parseTypeMap(hippo_bson_conversion_options_t *options, const Array &typemap
 			options->array_class_name = array_type;
 		}
 	}
+
+	if (typemap.exists(s_types)) {
+		if (!typemap[s_types].isArray()) {
+			throw MongoDriver::Utils::throwInvalidArgumentException("The 'types' key in the type map should be an array");
+		}
+
+		for (ArrayIter iter(typemap[s_types]); iter; ++iter) {
+			Variant key(iter.first());
+			String s_key = key.toString();
+
+			const Variant& data(iter.secondRef());
+			if (!data.isString()) {
+				throw MongoDriver::Utils::throwInvalidArgumentException("The typemap for type '" + s_key + "' should be a string");
```

Comment:
I think this exception can be improved, as "typemap" here doesn't jive with "type map" in the previous message. It may make more sense to rephrase as our other unexpected type errors (e.g. we expected a string and were given foo). That is something we can easily change later, though.